pkg/components/state/state_config.go

```diff
@@ -96,6 +100,8 @@ func GetOriginalStateKey(modifiedStateKey string) string {
 }
 
 func getStateConfiguration(storeName string) *StoreConfiguration {
+	statesConfigurationLock.Lock()
```

Comment:
Is it better to use `RWMutex` with `RLock` here?

Reply:
The statesConfiguration is also updated in this function. see: https://github.com/dapr/dapr/blob/master/pkg/components/state/state_config.go#L102

Reply:
Just saw that, thanks!